analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnModulus.cs

```diff
@@ -50,17 +59,16 @@ private static void VisitEquality(SyntaxNodeAnalysisContext c)
             if (CheckExpression(equalsExpression.Left, equalsExpression.Right, c.SemanticModel, out var constantValue)
                 || CheckExpression(equalsExpression.Right, equalsExpression.Left, c.SemanticModel, out constantValue))
             {
-                c.ReportIssue(Diagnostic.Create(Rule,
-                                                               equalsExpression.GetLocation(),
-                                                               constantValue < 0 ? "negative" : "positive"));
+                c.ReportIssue(Diagnostic.Create(Rule, equalsExpression.GetLocation(), constantValue < 0 ? "negative" : "positive"));
             }
         }
 
         private static bool CheckExpression(SyntaxNode node, ExpressionSyntax modulus, SemanticModel semanticModel, out int constantValue) =>
             ExpressionNumericConverter.TryGetConstantIntValue(node, out constantValue)
             && constantValue != 0
             && IsModulus(modulus)
-            && !IsUnsigned(modulus, semanticModel);
+            && !IsUnsigned(modulus, semanticModel)
```

Comment:
pro tip: now that you do this check, you can filter early based on the AST if it contains `Length` or `Count` before querying the semantic model

also, you seem to do 2 semantic model checks for the same `expression` - is it possible to do the job with just one?

Reply:
I wanted to keep them separate since they are logically checking 2 different things but I see the potential perf improvement.